File: onyx-database/src/com/onyx/persistence/annotations/IdentifierGenerator.java
package com.onyx.persistence.annotations;

/**
 * This enum is used to indicate what primary key generator to use when declared within the @Identifier annotation
 *
 * NONE - No auto generated primary key
 *
 * SEQUENCE - Sequential primary key generated key
 *
 * @author Tim Osborn
 *
 * @since 1.0.0
 *
 * <pre>
 *     <code>
 *          {@literal @}Identifier(generator = IdentifierGenerator.SEQUENCE)
 *          {@literal @}Attribute(nullable = false, size = 200)
 *           public long personID;
 *     </code>
 * </pre>
 *
 *
 * @see com.onyx.persistence.annotations.Identifier
 */
public enum IdentifierGenerator
{
    SEQUENCE,
    NONE;

    /**
     * Constructor
     */
    IdentifierGenerator()
    {

    }
}

File: onyx-database/src/main/kotlin/com/onyx/persistence/annotations/values/IdentifierGenerator.kt
package com.onyx.persistence.annotations.values

/**
 * This enum is used to indicate what primary key generator to use when declared within the @Identifier annotation
 *
 * NONE - No auto generated primary key
 *
 * SEQUENCE - Sequential primary key generated key
 *
 * @author Tim Osborn
 *
 * @since 1.0.0
 *
 * @Identifier(generator = IdentifierGenerator.SEQUENCE)
 * @Attribute(nullable = false, size = 200)
 * public long personID;
 *
 * @see com.onyx.persistence.annotations.Identifier
 */
enum class IdentifierGenerator {
    SEQUENCE,UUID,NONE
}
